### PR TITLE
For #24971 - Split full name from Address editor into first, middle and last name

### LINKED ISSUE #24971: Split full name in the Address Editor into First, Middle, and Last name
The AS Address data type currently expects first, middle, and last name instead of a single full name field. Rather than parsing the full name into its first, middle and last name, I have requested UX to provide designs that would align with desktop.

Screen Shot 2022-04-26 at 4 07 40 PM

This is pending designs.

### PR DESCRIPTION
**Pull request by Alexandru2909**
Fixes #24971

Address editor screen now uses first, middle and last name:

Screenshot_20220428_180758

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
We don't actually have designs for this yet. So, we should just wait until we advised on what the design and strings are.

**Comment by gabrielluong** [comment]
> @gabrielluong I would have wanted this to land before adding the "update address" [feature](https://github.com/mozilla-mobile/fenix/issues/24855) to map more values from the entity to the screen when displaying an existing address. But we can go either way.

Ok, let's land this immediately then because today is string freeze.

File: app/src/main/res/layout/fragment_address_editor.xml [hunk 1]
```diff
@@ -32,18 +32,106 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="16dp"
+            android:gravity="center_vertical"
+            android:letterSpacing="0.05"
+            android:paddingStart="3dp"
+            android:paddingEnd="0dp"
+            android:text="@string/addresses_middle_name"
+            android:textColor="?attr/textPrimary"
+            android:textSize="12sp"
+            android:labelFor="@id/card_number_input"
```
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_address_editor.xml:73]: These are pointing to the @id

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1568,8 +1568,12 @@
     <string name="addresses_add_address">Add address</string>
     <!-- Title of the "Manage addresses" screen -->
     <string name="addresses_manage_addresses">Manage addresses</string>
-    <!-- The header for the full name of an address -->
-    <string name="addresses_full_name">Full Name</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:1572]: This string needs to be marked for removedIn="101" https://github.com/mozilla-mobile/fenix/wiki/Removing-strings#marking-an-unused-string-to-be-removed

**Review by gabrielluong — APPROVED** [review]

**Comment by mcarare** [comment]
@gabrielluong I've edited the PR title. The issue number in the commit message is ok, just the branch was named incorrectly.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1569,7 +1569,13 @@
     <!-- Title of the "Manage addresses" screen -->
     <string name="addresses_manage_addresses">Manage addresses</string>
     <!-- The header for the full name of an address -->
-    <string name="addresses_full_name">Full Name</string>
+    <string name="addresses_full_name" moz:removedIn="102">Full Name</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:1572]: This also needs the ignore UnusedResource I believe

**Pull request merged**